src/iam/certmodules/pkcs11/pkcs11.cpp

```diff
@@ -18,6 +18,31 @@ namespace aos {
 namespace iam {
 namespace certhandler {
 
+/***********************************************************************************************************************
+ * Static
+ **********************************************************************************************************************/
+
+static Error IDToStr(const Array<uint8_t>& id, String& idStr)
+{
+    for (auto byte : id) {
+        auto err = idStr.PushBack('%');
+        if (!err.IsNone()) {
```

Comment:
as we allready know the required size of destination array I would suggest to have a single check at the beginning of the method

Reply:
You should not rely on implementation details. `PushBack` may return other than size overflow errors as well. Normally, if method returns error, it should be handled. And because `PushBack` also checks the size, no need for checking the size at the beginning.